runtime/near-vm-logic/src/gas_counter.rs

```diff
@@ -155,9 +154,8 @@ impl GasCounter {
         }
     }
 
-    pub fn pay_wasm_gas(&mut self, opcodes: u32) -> Result<()> {
-        let value = Gas::from(opcodes) * self.fast_counter.opcode_cost;
-        self.burn_gas(value)
+    pub fn pay_wasm_gas(&mut self, gas: Gas) -> Result<()> {
+        self.burn_gas(gas)
     }
```

Comment:
Consider in-lining this method. It loos like it can be called directly, with the same arguments.